ts/modules/full-screen.src.ts

```diff
@@ -1,80 +1,237 @@
 /* *
+ * (c) 2009-2020 Rafal Sebestjanski
  *
- *  (c) 2009-2020 Sebastian Bochann
+ * Full screen for Highcharts
  *
- *  Full screen for Highcharts
- *
- *  License: www.highcharts.com/license
- *
- *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
- *
- * */
+ * License: www.highcharts.com/license
+ */
 
+'use strict';
 import H from '../parts/Globals.js';
 
+const addEvent = H.addEvent,
+    Chart = H.Chart;
+
 /**
  * Internal types
  * @private
  */
 declare global {
     namespace Highcharts {
-        class FullScreen {
-            public constructor(container: HTMLDOMElement);
-            public init(container: HTMLDOMElement): void;
+        interface Chart {
+            fullscreenController: FullscreenController;
+            isFullscreen?: boolean;
+            unbindFullscreenEvent?: Function;
+        }
+        class FullscreenController {
+            public browserProps: {
+                fullscreenChange: (
+                    'fullscreenchange'|
+                    'mozfullscreenchange'|
+                    'webkitfullscreenchange'|
+                    'MSFullscreenChange'
+                );
+                requestFullscreen: (
+                    'msRequestFullscreen'|
+                    'mozRequestFullScreen'|
+                    'requestFullscreen'|
+                    'webkitRequestFullScreen'
+                );
+                exitFullscreen: (
+                    'exitFullscreen'|
+                    'mozCancelFullScreen'|
+                    'webkitExitFullscreen'|
+                    'msExitFullscreen'
+                );
+            };
+            public chart: Chart;
+            public closeFullscreen(): void;
+            public constructor(chart: Chart);
+            public openFullscreen(): void;
+            public setButtonText(): void;
+            public toggleFullscreen(): void;
         }
     }
 }
 
-/* eslint-disable no-invalid-this, valid-jsdoc */
-
 /**
- * The FullScreen class.
- * The module allows user to enable full screen mode in StockTools.
+ * The module allows user to enable display chart in full screen mode.
+ * Used in StockTools too.
  * Based on default solutions in browsers.
  *
- * @private
- * @class
- * @name Highcharts.FullScreen
+ */
+
+/**
+ * The flag is set to `true` if the chart is displayed in fullscreen mode.
  *
- * @param {Highcharts.HTMLDOMElement} container
- *        Chart container
+ * @name Highcharts.Chart#isFullscreen
+ * @type {boolean|undefined}
+ * @since next
  */
-var FullScreen = H.FullScreen = function (
-    this: Highcharts.FullScreen,
-    container: Highcharts.HTMLDOMElement
+
+/* eslint-disable no-invalid-this, valid-jsdoc */
+
+H.FullscreenController = function (
```

Comment:
Please write this in ES6 class pattern with doclets. That way you also avoid the `any` casts:
```ts
/**
 * ... Class description ...
 * @class
 * @name Highcharts.FullscreenController
 * @hideconstructor
 * @requires modules/exporting
 * @requires modules/full-screen
 */
class FullscreenControll {
    /* *
     *
     *  Constructors
     *
     * */
    /** @private */
    public constructor(chart: Highcharts.Chart) {
        this.chart = chart;
        this.browserProps = { ... };
    }
    /* *
     *
     *  Properties
     *
     * */
    /**
     * Chart managed by the fullscreen controller.
     * @name Highcharts.FullscreenController#chart
     * @type {Highcharts.Chart}
     */
    public chart: Highcharts.Chart;
    /** @private */
    public browserProps: Highcharts.FullscreenControll['browserProps'];
    /* *
     *
     *  Functions
     *
     * */
    /**
     * ... Description ...
     * @function Highcharts.FullscreenController.toggleFullscreen
     */
    public toggleFullscreen(): void {
    }
}
addEvent(Chart, 'beforeRender', function (this: Highcharts.Chart): void {
    /**
     * ...Description...
     * @name Highcharts.Chart#fullscreenController
     * @type {Highcharts.FullscreenController}
     * @requires modules/exporting
     * @requires modules/full-screen
     */
     this.fullscreenController = new FullscreenController(this);
 });
H.FullscreenController = FullscreenController;
export default H.FullscreenController;
```